kernelci.org/content/en/docs/api_pipeline/_index.md

```diff
@@ -0,0 +1,99 @@
+interface (Pub/Sub).  For example, when a new kernel revision is found and a
+new node is created for it, an event will be sent to tell subscribers about it
+with something like "A checkout node has been created for kernel revision
+v6.2-rc4".  The actual event is a [CloudEvents](https://cloudevents.io) object
+with some JSON data containing a subset of the corresponding node database
+entry.
+
+Any client code can subscribe to receive events with an API token and implement
+features based on how to handle these events.  The API generates events
+automatically whenever nodes are changing but clients may also use the
+interface to publish their own events and coordinate other parts of the
+pipeline too.
+
+### User Ownership
+
+Interacting with the API requires a token which is associated with a particular
+user.  Whenever a user sends some data such as a new node, it is owned by that
+user.  While all nodes are publicly readable, only the owner of the node can
+update it.  Users can also belong to groups to share data with other users.
+
+While the main KernelCI pipeline will be creating nodes with users from a
+particular `kernelci.org` group, all the other users can create their own data
+which will coexist in the database.  Then your own nodes can have parents
+created by users.  For example, you may submit test results that relate to a
+kernel build provided by KernelCI.
+
+## Pipeline
+
+The Pipeline is made up of all the client-side services that run the actual
+workloads that produces data and artifacts.  It's orchestrated based on events
+from the Pub/Sub interface and all the data is managed via the API.  For
+example, the pipeline is responsible for detecting new kernel revision,
+scheduling builds and tests, sending email reports and detecting regressions.
+However, any other service which has an API token is in fact part of the
+extended pipeline too.
+
+Pipeline services are also responsible for uploading various kinds of artifacts
+to some independent storage services and provide public URLs to access them.
+Artifacts typically include kernel source code tarballs, build artifacts, logs
+and test results in some raw format before they were submitted to the API.
+
+## Instances
+
+### Staging
+
+An instance has been set up on `staging.kernelci.org` for testing all pending
+changes.  The Docker logs are available in real-time via a [web
+interface](https://staging.kernelci.org:9088/) for both the API and the
+pipeline.  It also provides some [interactive API
+documentation](https://staging.kernelci.org:9000/latest/docs).  This instance
+is not stable, it's redeployed periodically with all open pull requests from
+GitHub merged together on a test integration branch.
+
+### Early Access
+
+In preparation for a full production roll-out, an [Early
+Access](/docs/api_pipeline/api/early-access) instance has been deployed in the Cloud (AKS)
+on `kernelci-api.eastus.cloudapp.azure.com`.  This is stable enough to let
```

Comment:
Thanks for relinking Early Access guide! Let's update instance links, too (these [were changed last month](https://github.com/kernelci/kernelci-deploy/commit/509dc39e9ea7db398ab82efd38fe2110b7fe0874#diff-18ad5a095d634e9316428cbebe628c23d55a6d08bd2d29aa127eb757e15528e7))

```suggestion
on `kernelci-api.westus3.cloudapp.azure.com`.  This is stable enough to let
```

Reply:
Fixed it. Thanks.